app/src/main/kotlin/pl/gov/mc/protego/ui/registration/RegistrationViewModel.kt

```diff
@@ -17,21 +18,19 @@ class RegistrationViewModel(
     private val msisdnValidator: MsisdnValidator,
     private val protegoServer: ProtegoServer,
     private val session: Session
-)  : ViewModel() {
+) : ViewModel() {
 
-    val msisdnError = MutableLiveData<String?>()
+    val msisdnError = MutableLiveData<Int?>()
```

Comment:
```suggestion
    private val _msisdnErrorResId = MutableLiveData<Int?>()
    val msisdnErrorResId : LiveData<Int?>
        get() = _msisdnErrorResId
```
dobrze jest określić w nazwie co to error int: czy kod błędu, resource id i trzeba będzie pobrać text do niego + trzeba zasłonić MutableLiveData poza klasą.

Reply:
Z nazwą masz rację, zmieniona.
Ale ten getter nie zadziała tak jak myślisz, getter co prawda ma w definicji typ zwracany `LiveData` ale i tak zwraca `msisdnErrorResId` które jest typu `MutableLiveData`, więc nadal każdy może sobie zrzutować to co getter zwrócił na `MutableLiveData`. Żeby temu zapobiec potrzebna byłaby tu bardziej wyrafinowana konstrukcja.

Reply:
chodzi tylko o fasadę tego wystawiasz co na zewnątrz, żeby klient `viewModel`u (w tym przypadku Activity) nie widział co się kryje w środku bez patrzenia u źródła. Zostawiając `MutableLiveData` dajesz przyzwolenie na settowanie go z zewnątrz, a tak jak ktoś zrzutuje `LiveData` to go na review cofniesz że nie powinien tak robić ;) Tu bardziej chodzi o czystość kodu niż blokadę na siłę ;)

Reply:
Może i masz rację, w ostateczności do końca nic nie da się ukryć, zawsze zostają jeszcze mechanizmy takie jak refleksja. Zmieniłem